lib/rex/post/mssql/ui/console.rb

```diff
@@ -26,8 +28,11 @@ def initialize(session, opts={})
             # The mssql client context
             self.session = session
             self.client = session.client
-            self.cwd = session.client.mssql_query('SELECT DB_NAME();')[:rows][0][0]
-            prompt = "%undMSSQL @ #{client.sock.peerinfo} (#{cwd})%clr"
+            # EnvChange Type 1: Database
+            db_env_change = session.client.initial_connection_info[:envs].select { |hash| hash[:type] == 1 }[0]
```

Comment:
Lets link the docs here https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-tds/2b3eb7e5-d43d-4d1b-bf4d-76b9e3afc791

Kinda surprised we don't have constants for these somewhere already, not a blocker for this PR but I think it's worth considering adding them

Reply:
Cross-referencing the original issue thread: https://github.com/rapid7/metasploit-framework/pull/18747#discussion_r1484234814

A set of namespaced constants that matches the table would be good

Reply:
Not a blocker; this breaks law of demeter IMO - maybe we could just add a convenience method for this lookup?

```
db_env_change = session.client.initial_connection_info(type: Mssql::Whatever::Makes::Sense::As::A::Constant::Database)
```